README.md

```diff
@@ -0,0 +1,219 @@
+      use its contents for an application-defined purpose.
+
+Their example is quite simple:
+
+== Server -> User Agent ==
+      Set-Cookie: SID=31d4d96e407aad42
+
+      == User Agent -> Server ==
+      Cookie: SID=31d4d96e407aad42
+
+In this example, the server responds to a request with the `Set-Cookie` header. This header sets the value of the `SID` cookie to `31d4d96e407aad42`.
+
+When the user visits another page on the same server, the browser sends the cookie back to the server, including the `Cookie: SID=31d4d96e407aad42` header in its request.
+
+Cookies are stored in the browser. The browser doesn't care about what's in the cookies you set. It just stores the data and sends it along on future requests to your server.
+
+## Using cookies
+
+So how would we use a cookie to store a reference to the user's shopping cart? Let's say that we create a cart the first time a user adds something to their cart. Then, in the response, we might include the header,
+
+      == Server -> User Agent ==
+      Set-Cookie: cart_id=273
+
+Only with the `cart.id` of the cart we just saved.
+
+When the user comes back to our site, their browser will include the cookie in their
+reply:
+
+      == User Agent -> Server ==
+      Cookie: cart_id=273
+
+We can look at this HTTP header, get the `cart_id` from it, and look it up using the `ActiveRecord` find method we know and love.
+
+## Security concerns
+
+Cookies are stored as plain text in a user's browser. Therefore, the user can see what's in them, and they can set them to anything they want.
+
+If you open the developer console in your browser, you can see the cookies set by the current site. In Chrome's console, you can find this under `Resources > Cookies`. You can delete any cookie you like. For example, if you delete your `user_session` cookie on `github.com`, you will find that you've been logged out.
+
+You can also edit cookies, for example with [this extension|edit_this_cookie].
+
+This presents a problem for us. If users can edit their `cart_id` cookie, then they can see other users' shopping carts.
+
+## Rails to the rescue
+
+Fortunately, Rails has a solution to this. When you set cookies in Rails, you usually don't manipulate the HTTP headers directly. Instead, you use the `session` method. The `session` method is available anywhere in the Rails response cycle, and it behaves like a hash:
+
+    # set cart_id
+    session[:cart_id] = @cart.id
+
+    # load the cart referenced in the session
+    @cart = session[:cart_id]
+
+You can store any simple Ruby object in the session. In fact, we don't need a `Cart` model at all—we can just store a list of items in the session!
+
+Rails manages all session data in a single cookie, named `_YOUR_RAILS_APP_NAME_session`. It *serializes* your all the key/value pairs you set with `session`, turning them into a big string. Whenever you set a `key` with the `session` method, Rails updates the value of its session cookie to this big string.
```

Comment:
Call out that serialize means to convert from a ruby object to a string representation